c7n/resources/appelb.py

```diff
@@ -292,8 +310,67 @@ def process(self, albs, event=None):
         return super(AppELBListenerFilter, self).process(albs, event)
 
     def __call__(self, alb):
-        listeners = self.listener_map[alb['LoadBalancerArn']]
-        return self.match(listeners)
+        matched = []
+        for listener in self.listener_map[alb['LoadBalancerArn']]:
+            if self.match(listener):
+                matched.append(listener)
+        if not matched:
+            return False
+        alb['c7n:MatchedListeners'] = matched
+        return True
+
+
+@actions.register('modify-listener')
+class AppELBModifyListenerPolicy(BaseAction):
+    """Action to modify the policy for an App ELB
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: appelb-modify-listener
+                resource: app-elb
+                filters:
+                  - type: listener
+                    key: Protocol
+                    value: HTTP
+                actions:
+                  - type: modify-listener
+                    protocol: HTTPS
+                    sslpolicy: "ELBSecurityPolicy-TLS-1-2-2017-01"
+                    certificate: "arn:aws:acm:region:123456789012:certificate/12345678-1234-1234-1234-123456789012"
+    """
+
+    schema = type_schema(
+        'modify-listener',
+        port={'type': 'integer'},
+        protocol={'enum': ['HTTP', 'HTTPS']},
+        sslpolicy={'type': 'string'},
+        certificate={'type': 'string'}
+    )
+
+    permissions = ("elasticloadbalancing:ModifyListener",)
+
+    def process(self, load_balancers):
+        args = {}
+        if 'port' in self.data:
+            args['Port'] = self.data.get('port')
+        if 'protocol' in self.data:
+            args['Protocol'] = self.data.get('protocol')
+        if 'sslpolicy' in self.data:
+            args['SslPolicy'] = self.data.get('sslpolicy')
+        if 'certificate' in self.data:
+            args['Certificates'] = [{'CertificateArn': self.data.get('certificate')}]
+        with self.executor_factory(max_workers=2) as w:
+            list(w.map(self.process_alb, load_balancers, [args]))
+
+    def process_alb(self, alb, args):
+        client = local_session(self.manager.session_factory).client('elbv2')
+        for matched_listener in alb['c7n:MatchedListeners']:
```

Comment:
this should probably use a alb.get(key, ()) if the filter was present. in general if there's an expected filter in a policy for an action to work, its good to assert that in the action validate method (where it can look at the rest of the policy).